engine/api/v2_workflow_run_craft.go

```diff
@@ -0,0 +1,469 @@
+	if err != nil {
+		return err
+	}
+	repo, err := repository.LoadRepositoryByVCSAndID(ctx, api.mustDB(), vcsServer.ID, run.RepositoryID)
+	if err != nil {
+		return err
+	}
+
+	u, err := user.LoadByID(ctx, api.mustDB(), run.UserID)
+	if err != nil {
+		return err
+	}
+
+	// Build run context
+	runContext := buildRunContext(*run, *vcsServer, *repo, *u)
+	run.Contexts = runContext
+
+	wref := WorkflowRunEntityFinder{
+		run:                    *run,
+		runRepo:                *repo,
+		runVcsServer:           *vcsServer,
+		actionsCache:           make(map[string]sdk.V2Action),
+		workerModelCache:       make(map[string]sdk.V2WorkerModel),
+		repoCache:              make(map[string]sdk.ProjectRepository),
+		vcsServerCache:         make(map[string]sdk.VCSProject),
+		repoDefaultBranchCache: make(map[string]string),
+		userName:               u.Username,
+	}
+
+	// Retrieve all deps
+	for jobID := range run.WorkflowData.Workflow.Jobs {
+		j := run.WorkflowData.Workflow.Jobs[jobID]
+
+		// Get worker model
+		completeName, msg, err := wref.searchEntity(ctx, api.mustDB(), api.Cache, j.WorkerModel, sdk.EntityTypeWorkerModel)
+		if err != nil {
+			log.ErrorWithStackTrace(ctx, err)
+			return stopRun(ctx, api.mustDB(), run, &sdk.V2WorkflowRunInfo{
+				WorkflowRunID: run.ID,
+				Level:         sdk.WorkflowRunInfoLevelError,
+				Message:       "unable to trigger workflow. Please contact an administrator",
+			})
+		}
+		if msg != nil {
+			return stopRun(ctx, api.mustDB(), run, msg)
+		}
+		j.WorkerModel = completeName
+
+		// Get actions and sub actions
+		msg, err = searchActions(ctx, api.mustDB(), api.Cache, &wref, j.Steps)
+		if err != nil {
+			log.ErrorWithStackTrace(ctx, err)
+			return stopRun(ctx, api.mustDB(), run, &sdk.V2WorkflowRunInfo{
+				WorkflowRunID: run.ID,
+				Level:         sdk.WorkflowRunInfoLevelError,
+				Message:       fmt.Sprintf("unable to retrieve job[%s] definition. Please contact an administrator", jobID),
+			})
+		}
+		if msg != nil {
+			return stopRun(ctx, api.mustDB(), run, msg)
+		}
+
+	}
+
+	for k, v := range wref.actionsCache {
+		run.WorkflowData.Actions = make(map[string]sdk.V2Action)
+		run.WorkflowData.Actions[k] = v
+	}
+	for k, v := range wref.workerModelCache {
+		run.WorkflowData.WorkerModels = make(map[string]sdk.V2WorkerModel)
+		run.WorkflowData.WorkerModels[k] = v
+	}
+
+	tx, err := api.mustDB().Begin()
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback() // nolint
+
+	run.Status = sdk.StatusBuilding
+	if err := workflow_v2.UpdateRun(ctx, tx, run); err != nil {
+		return err
+	}
+
+	enqueueRequest := sdk.V2WorkflowRunEnqueue{
+		RunID:  run.ID,
+		UserID: run.UserID,
+	}
+
+	select {
+	case api.workflowRunTriggerChan <- enqueueRequest:
+		log.Debug(ctx, "workflow run %s %d trigger in chan", run.WorkflowName, run.RunNumber)
+	default:
+	}
+	if err := api.Cache.Enqueue(workflow_v2.WorkflowEngineKey, enqueueRequest); err != nil {
```

Comment:
this should be in the "defaut" section, no ?

Reply:
yes